Softeq.XToolkit.WhiteLabel/Bootstrapper/AutofacContainerBuilder.cs

```diff
@@ -0,0 +1,147 @@
+// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using System.Collections.Generic;
+using Autofac;
+using Autofac.Builder;
+using Softeq.XToolkit.Common.Extensions;
+using Softeq.XToolkit.WhiteLabel.Bootstrapper.Abstract;
+using IContainer = Softeq.XToolkit.WhiteLabel.Bootstrapper.Abstract.IContainer;
+
+namespace Softeq.XToolkit.WhiteLabel.Bootstrapper
+{
+    public class AutofacContainerBuilder : IContainerBuilder
+    {
+        private readonly ContainerBuilder _builder;
+        private readonly List<Action<IContainer>> _buildActions;
+
+        public AutofacContainerBuilder()
+        {
+            _builder = new ContainerBuilder();
+            _buildActions = new List<Action<IContainer>>();
+        }
+
+        public IConcreteRegistration PerDependency<T1, T2>() where T1 : T2
+        {
+            return new AutofacRegistration<T1>(_builder.RegisterType<T1>().As<T2>().InstancePerDependency());
+        }
+
+        public IConcreteRegistration PerDependency<T1>()
+        {
+            return new AutofacRegistration<T1>(_builder.RegisterType<T1>().InstancePerDependency());
+        }
+
+        public IConcreteRegistration PerDependency<T1>(Func<IContainer, T1> func)
+        {
+            return new AutofacRegistration<T1>(_builder.Register(c => func.Invoke(c.Resolve<IContainer>()))
+                .InstancePerDependency());
+        }
+
+        public IConcreteRegistration PerDependency(Type type)
+        {
+            return new AutofacRegistration(_builder.RegisterType(type).InstancePerDependency());
+        }
+
+        public IConcreteRegistration PerDependency<T1>(Func<IContainer, object> func)
+        {
+            return new AutofacRegistration<T1>(_builder.Register(c => func.Invoke(c.Resolve<IContainer>())).As<T1>()
+                .InstancePerDependency());
+        }
+
+        public IConcreteRegistration Singleton<T1, T2>() where T1 : T2
+        {
+            return new AutofacRegistration<T1>(_builder.RegisterType<T1>().As<T2>().InstancePerLifetimeScope().SingleInstance());
```

Comment:
Looks like InstancePerLifetimeScope().SingleInstance() will do the same as simply SingleInstance(), perhaps this should be simplified

Reply:
fixed